synapse/module_api/__init__.py

```diff
@@ -72,6 +72,16 @@ def public_room_list_manager(self):
         """
         return self._public_room_list_manager
 
+    def get_user_by_id(self, user_id: str) -> Optional[Dict[str, Any]]:
+        """Get user by user_id
+
+        Args:
+            user_id: Fully qualified user id.
+        Returns:
+            the user dictionary or None if not found.
```

Comment:
I'm a bit concerned about returning an unstructured dict here, with no contract about what fields it contains. What assumptions is the module allowed to make?

Reply:
Do we have a user type that could be constructed?

Reply:
Not at present. You could define one (maybe in the `types` module, using `attr.s`), or just list the fields here.

Reply:
Extremely sorry to let this sit for such a long time. I reviewed it's still useful with the new modules architecture. There is now a `User` type. It tries to match some of the way `Requester` names things, for example using `user` instead of `user_id` or `name` (actual database column). I was tempted to call it `ModuleApiUser` to match some of the specific types found, but since it pretty much matches what comes out of the database, left it as `User`.